mcstatus/protocol/connection.py

```diff
@@ -12,7 +13,10 @@
 
 from ..scripts.address_tools import ip_type
 
-BytesConvertable = Union[SupportsIndex, Iterable[SupportsIndex]]
+if TYPE_CHECKING:
+    from typing_extensions import SupportsIndex  # Python 3.7 doesn't support this yet.
+
+    BytesConvertable = Union[SupportsIndex, Iterable[SupportsIndex]]
```

Comment:
Because `BytesConvertable` is a runtime variable, and not an annotation, moving its declaration out of the if block will raise an error at runtime because `SupportsIndex` isn't defined at runtime. This is why I moved the declaration inside the if block, but I can move it back out and make `SupportsIndex` a string.

Reply:
Oh, yeah I didn't realize `__future__.annotations` didn't cover that, in hindsight it's obvious that it won't. Well, it doesn't really matter where it is then, I personally don't mind type declarations as strings, but since this PR is trying to avoid that as much as possible, it's probably better to just keep it in the if.